Tests/AsyncHTTPClientTests/HTTPClientNIOTSTests.swift

```diff
@@ -74,19 +74,24 @@ class HTTPClientNIOTSTests: XCTestCase {
     func testConnectionFailError() {
         guard isTestingNIOTS() else { return }
         let httpBin = HTTPBin(ssl: true)
-        let httpClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup))
+        let httpClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup),
+                                    configuration: .init(timeout: .init(connect: .milliseconds(100),
+                                                                        read: .milliseconds(100))))
+
         defer {
             XCTAssertNoThrow(try httpClient.syncShutdown(requiresCleanClose: true))
         }
+
         let port = httpBin.port
         XCTAssertNoThrow(try httpBin.shutdown())
 
-        do {
-            _ = try httpClient.get(url: "https://localhost:\(port)/get").wait()
-            XCTFail("This should have failed")
-        } catch ChannelError.connectTimeout {
-        } catch {
-            XCTFail("Error should have been ChannelError.connectTimeout not \(type(of: error))")
+        XCTAssertThrowsError(try httpClient.get(url: "https://localhost:\(port)/get").wait()) { error in
+            switch error {
+            case ChannelError.connectTimeout(let timeout):
```

Comment:
@weissi do you know if there is a safer pattern for this case, where enum has an associated value?

Reply:
What don't you like about this pattern?

Reply:
This is potentially dangerous, if you miss the `XCTFail` call in the default clause, you might miss when the test breaks. This is why I'm replacing `do-try-XCTFail-catch` pattern with `XCTAssertThrowsError`

Reply:
I see what you mean. No, there isn't really a better pattern here sadly.

Reply:
I usually do this

```
XCTAssertEqul(.connectTimeout(.milliSeconds(150)), error as? ChannelError)
```

as it's much safer